lib/g_searcher/reports.ex

```diff
@@ -0,0 +1,89 @@
+defmodule GSearcher.Reports do
+  import Ecto.Query, warn: false
+
+  alias Ecto.Multi
+  alias GSearcher.{Repo, SearchResults}
+  alias GSearcher.SearchResults.Report
+
+  NimbleCSV.define(CSVParser, separator: "\t", escape: "\"")
+
+  def create_report(user_id, title, file_path) do
+    Multi.new()
+    |> Multi.insert(
+      :report,
+      Report.create_changeset(%{title: title, csv_path: file_path, user_id: user_id})
+    )
+    |> Multi.run(:search_keywords, fn _, %{report: %{id: report_id, csv_path: csv_path}} ->
+      save_keywords_from_file(csv_path, report_id)
+    end)
+    |> Repo.transaction()
+    |> handle_report_transaction_response()
+  end
+
+  defp save_keywords_from_file(csv_path, report_id) do
+    with {:ok, csv_path} <- validate_file_path(csv_path),
```

Comment:
I feel like the error is only from `validate_file_path` so should we somehow do if-else on this condition and put the rest inside. Not sure... 🤔